lib/bridge/IrcHandler.js

```diff
@@ -882,12 +892,16 @@ IrcHandler.prototype._onPrivateMode = Promise.coroutine(function*(req, server, c
                 return this._setMatrixRoomAsInviteOnly(room, enabled);
             default:
                 // Not reachable, but warn anyway in case of future additions
-                log.warn(`onMode: Unhandled channel mode ${mode}`);
+                req.log.warn(`onMode: Unhandled channel mode ${mode}`);
                 return Promise.resolve();
         }
     });
 
-    yield Promise.all(promises);
+    yield Promise.all(promises).then(() => {
+        matrixRooms.forEach((room) => {
+            this.ircBridge.getStore().setModeForRoom(room.getId(), mode, enabled);
```

Comment:
`k` and `i` aren't caught here.

Reply:
why not have the block set the modes on the room as early as ~line 860 (after the "no mapped rooms" check)?